### DISCUSSION TITLE
Questions about backup schedule

### DISCUSSION THREAD

**Discussion by ovizii** [garethgeorge/backrest#332] [Q&A] [answered]
I switched from cron based to "Max Frequency Days" or "Max Frequency Hours" in the hopes that it would be executed the number of days/hours after the last run, but it ignored the previous cron triggered backups and scheduled the next one 24h after I had saved the job. 

So the problem is, how do I make a plan execute say early morning without having to get up at 3am and saving the plan?

**Comment by garethgeorge** [comment]
If you want a backup to run at a specific time of day cron is the right solution, max frequency hours and max frequency days are for running an operation at a specified interval when you don't care about the exact time of execution.

Prune and check operations are always scheduled relative to the last time the operation ran.

Backup scheduling is somewhat special in that it is always relative to the current time -- this is to ensure you don't get a burst of operations running as soon as you boot up your computer if you've skipped a few backups, but I'm definitely open to discussion re: whether this is the right policy. So for max frequency hours / max frequency days this will always be after startup or config change. Cron behaves more expectedly for these backup operations.

**Comment by ovizii** [comment]
Hm, maybe I'm complicating things. Can you advise on the best solution? 

At the moment I have 2 jobs which should run daily and one which should run weekly. 

The following additional criteria apply:

- they should start after 3am as the HDs are audible from where I sleep, and I don't want backups to hog my bandwidth during the day.

The backup durations vary. Sometimes its a matter of minutes sometimes up to an hour. I monitor my backups with healthchecks. 
I need the backups to trigger a curl command after successfully completing. Backrest does all of this beautifully, apart from the scheduling I am struggling with.

On the healthchecks side, I can choose a schedule when to expect a successful completion trigger. The problem is, since I don't know when they'll finish, I at least need to know when they start.

![image](https://github.com/garethgeorge/backrest/assets/837902/034c3724-f0a9-4d90-ad8c-79adf8bbdeec)
or alternatively
![image](https://github.com/garethgeorge/backrest/assets/837902/65603b20-f8f1-4d78-a00d-00879cfcd0d0)

Anyway, I guess this weekend I am staying up till 3m and schedule them all as "Max Frequency Days" or "Max Frequency Hours" and set their schedule in healthchecks to every 7 days, grace 4h and every 24h grace 2h. That should do the trick.

**Comment by garethgeorge** [comment, marked as answer]
One good option re: healthchecks is to proactively indicate failure e.g. per the example here https://garethgeorge.github.io/backrest/cookbooks/command-hook-examples#notify-a-healthcheck-service

You can then set a quite large timeout like 24 hours or more to catch cases where backrest itself crashes (which should be very rare!).

**Comment by ovizii** [comment]
Thanks for the sample script. Until now, I was triggering a successful backup with the Condition_Snapshot_Success and if that didn't happen, it was obviously a fail. 

I will look into your example, that looks nice. The `.Error` does not seem to be documented but it's obvious. Very nice documentation otherwise: https://garethgeorge.github.io/backrest/docs/hooks

I will peruse the rest of the templating documentation and figure out a solution for myself :-)

**Comment by ovizii** [comment]
Please excuse me, but I'd like to add another cron related question. 
In backrest's GUI I can't seem to be able to select `0 5 * * SUN` — the closest I could get to that was `0 5 */7 * *` — is it be being blonde or is this the way it is at the moment?

**Reply by garethgeorge** [reply]
just saw this never got an answer, should be possible but may have been broken at the time you were looking!

<img width="539" alt="image" src="https://github.com/user-attachments/assets/9f2d0822-550c-4460-ab43-0337c7ed53cf">

**Comment by inode64** [comment]
Can Healthchecks hooks be included natively?
In healtcheks you can be informed of the number of files, byte recorded

Greetings and good work

**Reply by garethgeorge** [reply]
Hey, Backrest's command cookbook shows a few examples that work today with healthchecks.io https://garethgeorge.github.io/backrest/cookbooks/command-hook-examples#notify-a-healthcheck-service . It may also be possible to expand that example with something that reports metrics etc if that's supported

I'm curious what would you be wanting out of a native integration?

**Reply by inode64** [reply]
It was like the other Discord, Slack, Shoutrrr or Gotify hooks
Have the data sent to the integrated healthchecks service
